tests/path_gateway_dag_test.go

```diff
@@ -563,10 +669,168 @@ func TestNativeDag(t *testing.T) {
 					Contains("</html>"),
 				),
 			},
+			helpers.SingleRangeTestTransform(t,
+				SugarTest{
+					Name: Fmt("GET {{name}} on /ipfs with no explicit header and single range", row.Name),
+					Request: Request().
+						Path("/ipfs/{{cid}}/", dagTraversalCID).
+						Headers(
+							Header("Range", "bytes=6-16"),
+						),
+					Response: Expect().
+						Headers(
+							Header("Content-Type", "application/vnd.ipld.dag-{{format}}", row.Format),
+						),
+				},
+				helpers.SimpleByteRange(6, 16, dagTraversal.RawData()[6:17]),
+				dagTraversal.RawData(),
+			),
+			helpers.SingleRangeTestTransform(t,
+				SugarTest{
+					Name: Fmt("GET {{name}} on /ipfs with dag content headers and single range", row.Name),
+					Request: Request().
+						Path("/ipfs/{{cid}}/", dagTraversalCID).
+						Headers(
+							Header("Accept", "application/vnd.ipld.dag-{{format}}", row.Format),
+							Header("Range", "bytes=6-16"),
+						),
+					Response: Expect().
+						Headers(
+							Header("Content-Type", "application/vnd.ipld.dag-{{format}}", row.Format),
+						),
+				},
+				helpers.SimpleByteRange(6, 16, dagTraversal.RawData()[6:17]),
+				dagTraversal.RawData(),
+			),
+			helpers.SingleRangeTestTransform(t,
+				SugarTest{
+					Name: Fmt("GET {{name}} on /ipfs with non-dag content headers and single range", row.Name),
+					Request: Request().
+						Path("/ipfs/{{cid}}/", dagTraversalCID).
+						Headers(
+							Header("Accept", "application/{{format}}", row.Format),
+							Header("Range", "bytes=6-16"),
+						),
+					Response: Expect().
+						Headers(
+							Header("Content-Type", "application/{{format}}", row.Format),
+						),
+				},
+				helpers.SimpleByteRange(6, 16, dagTraversal.RawData()[6:17]),
+				dagTraversal.RawData(),
+			),
 		}
 
 		RunWithSpecs(t, tests, specs.PathGatewayDAG)
 	}
+
+	dagCborFixture := car.MustOpenUnixfsCar("path_gateway_dag/dag-cbor-traversal.car").MustGetRoot()
+	dagCborCID := dagCborFixture.Cid()
+	var dagJsonConvertedData []byte
+	RunWithSpecs(t, SugarTests{
+		SugarTest{
+			Name: "Convert application/vnd.ipld.dag-cbor to application/vnd.ipld.dag-json",
+			Hint: "",
+			Request: Request().
+				Path("/ipfs/{{cid}}/", dagCborCID).
+				Headers(
+					Header("Accept", "application/vnd.ipld.dag-json"),
+				),
+			Response: Expect().Body(Checks("", func(t []byte) bool {
+				innerCheck := IsJSONEqual(dagCborFixture.Formatted("dag-json")).Check(t)
+				if innerCheck.Success {
+					dagJsonConvertedData = t
+					return true
+				}
+				return false
+			})),
+		},
+	}, specs.PathGatewayDAG)
+
+	if dagJsonConvertedData != nil {
```

Comment:
@laurentsenta if there's an error here and this ends up being nil the number of tests might not match exactly between runs against different gateways. I copied this pattern from the prior range tests. Is this ok?

Reply:
To me that's acceptable: we should make sure the tooling / framework maintains the counts,
That code is user-side, and it's a bandaid until we solve https://github.com/ipfs/gateway-conformance/issues/98, the simpler we keep it, the easier to replace it.